### PR TITLE
feat(rc): Add Remote Config Parameter Value Type support

### PR DESCRIPTION
**Pull request by lahirumaramba**
go/admin-sdk-rc-parameter-value-types

- Add RC Parameter Value Type.
- Update unit tests.
~- Integration tests will be added following the REST API launch.~
~- Added `release:stage` to trigger existing integration tests (to test backward compatibility).~
~- `Do not merge` until the BE is updated.~

- Update integration tests.

RELEASE NOTE: Added Remote Config Parameter Value Type Support.

### CODE REVIEW

**Review by egilmorez — APPROVED** [review]
LGTM, thanks!

File: src/test/java/com/google/firebase/remoteconfig/TemplateTest.java [hunk 1]
```diff
@@ -49,11 +49,13 @@
...
                   .setDefaultValue(ParameterValue.inAppDefault())
                   .setDescription("greeting header text")
-                  .setConditionalValues(CONDITIONAL_VALUES),
+                  .setConditionalValues(CONDITIONAL_VALUES)
+                  .setValueType(ParameterValueType.STRING),
           "greeting_text", new Parameter()
                   .setDefaultValue(ParameterValue.inAppDefault())
                   .setDescription("greeting text")
                   .setConditionalValues(CONDITIONAL_VALUES)
+                  .setValueType(ParameterValueType.STRING)
```
Comment (hiranya911) [inline src/test/java/com/google/firebase/remoteconfig/TemplateTest.java:58]: Do we have coverage here for parameters without types?

**Review by hiranya911 — APPROVED** [review]
Thanks! LGTM with a couple of nits.

File: src/test/java/com/google/firebase/remoteconfig/TemplateTest.java (hunk 1 above)
Comment (lahirumaramba) [reply src/test/java/com/google/firebase/remoteconfig/TemplateTest.java:58]: Good call. Updated to have a parameter without a type.

**Pull request merged**